javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -552,7 +557,20 @@ protected void generatePagedAsyncSinglePage(ClientMethod clientMethod, JavaType
                 AddOptionalAndConstantVariables(function, clientMethod, restAPIMethod.getParameters(), settings);
                 ApplyParameterTransformations(function, clientMethod, settings);
                 ConvertClientTypesToWireTypes(function, clientMethod, restAPIMethod.getParameters(), clientMethod.getClientReference(), settings);
-                String restAPIMethodArgumentList = String.join(", ", clientMethod.getProxyMethodArguments(settings));
+
+                List<String> serviceMethodArgs = clientMethod.getProxyMethodArguments(settings)
+                        .stream()
+                        .map(argVal -> {
+                            if(clientMethod.getParameters().stream().filter(param -> param.getName().equals(argVal))
+                                    .anyMatch(param -> clientMethod.getMethodTransformationDetails().stream()
+                                            .anyMatch(transformation -> param.getName().equals(transformation.getOutParameter().getName())))) {
+                                return argVal + "Local";
+                            }
+                            return argVal;
+                        })
+                        .collect(Collectors.toList());
```

Comment:
Is all these duplicate code? If so can it be moved to a shared method?

Reply:
Moved to a method.